### PR TITLE
Vtf doc

### LINKED ISSUE #1807: Doc: VTF writer

### PR DESCRIPTION
**Pull request by davidsean**
Fixes #1807

Description of changes:
 - updated text from TCL
 - made VTF id's continuous and increasing
 - added map between VTF and espresso PID

PR Checklist
------------
 - [x] Tests?
   - [x] Interface
   - [ ] Core 
 - [x] Docs?

### CODE REVIEW

File: doc/sphinx/io.rst [hunk 1]
```diff
@@ -102,8 +102,8 @@ To give an example::
     checkpoint.register("skin")
 
     system = espressomd.System()
-    # ... set system properties like box_l or
-    timestep here ... checkpoint.register("system")
+    # ... set system properties like box_l or timestep here ...
```
Comment (KaiSzuttor) [inline doc/sphinx/io.rst]: time_step

File: doc/sphinx/io.rst [hunk 2]
```diff
@@ -265,27 +268,55 @@ structure block and at least one coordinate block is required.
...
 VCF format contain at least one coordinate block, while files in the VTF
-format contain a single structure block first and an arbitrary number of
-coordinate blocks afterwards, thus allowing to store all information for
+format contain a single structure block (usually as a header) and an arbitrary number of
+coordinate blocks (time frames) afterwards, thus allowing to store all information for
 a whole simulation in a single file. For more details on the format,
-refer to the homepage of the format .
+refer to the VTF homepage (https://github.com/olenz/vtfplugin/wiki).
+
+Creating files in these formats from within is supported by the commands ``writevsf``
```
Comment (KaiSzuttor) [inline doc/sphinx/io.rst]: please link to the in-source documentation with ```:meth:`espressomd.io.writevsf` ``` or similar

File: doc/sphinx/io.rst [hunk 3]
```diff
@@ -265,27 +268,55 @@ structure block and at least one coordinate block is required.
...
-given Tcl channel. To create a VTF file, first use at the beginning of
-the simulation, and then ``writevcf`` after each timestep to generate a
-trajectory of the whole simulation.
+A standalone VTF file can simply be 
+
+.. code:: python
+
+    import espressomd
+    from espressomd.io.writer import vtf
+    system = espressomd.System()
```
Comment (KaiSzuttor) [inline doc/sphinx/io.rst]: please use the new syntax for setting up a system with the box_l parameter

File: doc/sphinx/io.rst [hunk 4]
```diff
@@ -295,75 +326,62 @@ timesteps. This is a restriction of VMD itself, not of the format.
...
+
+Writes a structure block describing the system’s structure to the given channel.
+for example
 
-writevsf(fp,types)
+.. code:: python
 
+    import espressomd
+    from espressomd.io.writer import vtf
+    system = espressomd.System()
```
Comment (KaiSzuttor) [inline doc/sphinx/io.rst]: please use new syntax

File: doc/sphinx/io.rst [hunk 5]
```diff
@@ -295,75 +326,62 @@ timesteps. This is a restriction of VMD itself, not of the format.
...
-
-Specify the coordinates of which particles should be written. If `types` is
-used, all coordinates will be written (in the ordered timestep format).
-Otherwise, has to be a Tcl-list specifying the pids of the particles.
-The default is `types="all"`. 
-Example
-`pids =[0, 23, 42]`
-`pids="all"`
-
 .. _writevcf\: Writing the coordinates:
```
Comment (KaiSzuttor) [inline doc/sphinx/io.rst:346]: does this link work without the ticks?

File: src/python/espressomd/io/writer/vtf.py [hunk 6]
```diff
@@ -1,3 +1,32 @@
+def vtf_pid_map(system, types='all'):
+    """
+    Generates a VTF particle index map to ESPResSo ``id``.
+    This fills the gap for particle ID's as required by VMD
+    
+    Parameters
+    ----------
+    system: espressomd.System() object
+    types : str
```
Comment (KaiSzuttor) [inline src/python/espressomd/io/writer/vtf.py]: please use ```:obj:`str` ```

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: doc/sphinx/io.rst (hunk 5 above)
Comment (davidsean) [reply doc/sphinx/io.rst:346]: I checked and it seems fine

**Comment by KaiSzuttor** [comment]
something seems to be wrong with python3 compatibility

**Comment by RudolfWeeber** [comment]
Once Py3 is sorted out, this can be merged.

**Pull request merged**